spec/ddtrace/runtime/metrics_spec.rb

```diff
@@ -4,250 +4,37 @@
-
-          allow(runtime_metrics).to receive(:gauge)
-        end
-
-        context 'with Ruby < 3', if: RUBY_VERSION < '3.0.0' do
-          it 'records global cache counters' do
-            flush
-
-            expect(runtime_metrics).to have_received(:gauge)
-              .with(Datadog::Ext::Runtime::Metrics::METRIC_GLOBAL_CONSTANT_STATE, kind_of(Numeric))
-              .once
-
-            expect(runtime_metrics).to have_received(:gauge)
-              .with(Datadog::Ext::Runtime::Metrics::METRIC_GLOBAL_METHOD_STATE, kind_of(Numeric))
-              .once
-          end
-        end
-
-        context 'with Ruby >= 3', if: RUBY_VERSION >= '3.0.0' do
-          it 'records constant cache counter' do
-            flush
-
-            expect(runtime_metrics).to have_received(:gauge)
-              .with(Datadog::Ext::Runtime::Metrics::METRIC_GLOBAL_CONSTANT_STATE, kind_of(Numeric))
-              .once
-          end
-        end
-      end
-    end
-
-    it_behaves_like 'a flush of all runtime metrics'
-  end
-
-  describe '#gc_metrics' do
-    subject(:gc_metrics) { runtime_metrics.gc_metrics }
-
-    context 'on MRI' do
-      before { skip unless PlatformHelpers.mri? }
-
-      it 'has a metric for each value in GC.stat' do
-        is_expected.to have(GC.stat.keys.size).items
-
-        gc_metrics.each do |metric, value|
-          expect(metric).to start_with(Datadog::Ext::Runtime::Metrics::METRIC_GC_PREFIX)
-          expect(value).to be_a_kind_of(Numeric)
-        end
-      end
-    end
-
-    context 'on JRuby' do
-      before { skip unless PlatformHelpers.jruby? }
-
-      it 'has a metric for each value in GC.stat' do
-        is_expected.to have_at_least(GC.stat.keys.count).items
-
-        gc_metrics.each do |metric, value|
-          expect(metric).to start_with(Datadog::Ext::Runtime::Metrics::METRIC_GC_PREFIX)
-          expect(value).to be_a_kind_of(Numeric)
-        end
-      end
-    end
-  end
-
-  describe '#default_metric_options' do
-    subject(:default_metric_options) { runtime_metrics.default_metric_options }
-
-    describe ':tags' do
-      subject(:default_tags) { default_metric_options[:tags] }
-
-      context 'when no services have been registered' do
-        it do
-          is_expected.to include(*Datadog::Core::Metrics::Client.default_metric_options[:tags])
-          is_expected.to include('language:ruby')
-        end
-      end
-
-      context 'when services have been registered' do
-        let(:services) { %w[parser serializer] }
-
-        before { services.each { |service| runtime_metrics.register_service(service) } }
-
-        it do
-          is_expected.to include(*Datadog::Core::Metrics::Client.default_metric_options[:tags])
-          is_expected.to include('language:ruby')
-          is_expected.to include(*services.collect { |service| "service:#{service}" })
-        end
+        expect(Datadog.runtime_metrics.metrics.default_metric_options[:tags]).to_not include("service:#{service}")
```

Comment:
These tests seem a bit weird, because they're not testing the behavior of `Datadog::Runtime::Metrics`, but of `Datadog::Core::Runtime::Metrics`.

If `Datadog::Runtime::Metrics` is effectively stays a shim for the rest of the API, I suggest testing it in a similar way as we did for `Datadog::Tracing`: check that it calls the components in the expected ways, but not directly test those components (such tests would be part of the specific components' tests).

Reply:
True; it probably should assert it sent the message only. `Core::Runtime::Metrics` should check the tags it sets instead.